ehrapy/anndata/anndata_ext.py

```diff
@@ -154,63 +154,130 @@ def anndata_to_df(
-    will be stored as boolean columns whereas the other non numerical columns will be stored as category.
+def move_to_obs(adata: AnnData, to_obs: list[str] | str, copy_obs: bool = False) -> AnnData:
+    """Move inplace or copy features from X to obs. Note that columns containing boolean values (either 0/1 or True(
+    true)/False(false)) will be stored as boolean columns whereas the other non numerical columns will be stored as
+    category.
 
     Args:
         adata: The AnnData object
         to_obs: The columns to move to obs
-        copy: Whether to return a copy or not
+        copy_obs: The values are copied to obs (and therefore kept in X) instead of moved completely
 
     Returns:
-        The original AnnData object with moved columns from X to obs
+        The original AnnData object with moved or copied columns from X to obs
     """
-    if copy:
-        adata = adata.copy()
 
     if isinstance(to_obs, str):  # pragma: no cover
         to_obs = [to_obs]
+
     # don't allow moving encoded columns as this could lead to inconsistent data in X and obs
     if any(column.startswith("ehrapycat") for column in to_obs):
         raise ObsMoveError(
             "Cannot move encoded columns from X to obs. Either undo encoding or remove them from the list!"
         )
-    indices = adata.var_names.isin(to_obs)
-    df = adata[:, indices].to_df()
-    adata._inplace_subset_var(~indices)
-    adata.obs = adata.obs.join(df)
-    updated_num_uns, updated_non_num_uns, num_var = _update_uns(adata, to_obs)
-    # cast numerical values from object
-    adata.obs[num_var] = adata.obs[num_var].apply(pd.to_numeric, errors="ignore", downcast="float")
-    # cast non numerical values from object to either bool (if possible) or category
-    adata.obs = _cast_obs_columns(adata.obs)
-    adata.uns["numerical_columns"] = updated_num_uns
-    adata.uns["non_numerical_columns"] = updated_non_num_uns
 
-    if copy:
-        return adata
+    if not all(elem in adata.var_names.values for elem in to_obs):
+        raise ValueError(f"Columns `{[x for x in to_obs if x not in adata.var_names.values]}` are not in var_names.")
+
+    if copy_obs:
+        indices = adata.var_names.isin(to_obs)
+        df = adata[:, indices].to_df()
+        adata.obs = adata.obs.join(df)
+        num_set = set(adata.uns["numerical_columns"].copy())
+        non_num_set = set(adata.uns["non_numerical_columns"].copy())
+        var_num = []
+        var_non_num = []
+        for var in to_obs:
+            if var in num_set:
+                var_num.append(var)
+            elif var in non_num_set:
+                var_non_num.append(var)
+        # cast numerical values from object
+        adata.obs[var_num] = adata.obs[var_num].apply(pd.to_numeric, errors="ignore", downcast="float")
+        # cast non numerical values from object to either bool (if possible) or category
```

Comment:
Why? Else, the comment is more or less useless because the code line actually says what it does